src/Substance.hs

```diff
@@ -378,34 +400,28 @@ applySubstHelper sigma (AT t) = AT (applySubst sigma t)
 subst :: VarEnv -> M.Map Y Arg -> [K] -> [K] -> M.Map Y Arg
 subst varEnv sigma argTypes formalTypes = let types = zip argTypes formalTypes
                                               sigma2 = foldl (substHelper varEnv) sigma types
-                                          in if ((length argTypes) /= (length formalTypes))
-                                            then error ("Arguments list lengths are not equal")
+                                          in if length argTypes /= length formalTypes
+                                            then error "Arguments list lengths are not equal"
```

Comment:
Give more descriptive error message: "There were $NUM1 arguments to the constructor $CTOR, but it expects $NUM2 arguments"